### PR TITLE
Improving dencoder coverage

### PR DESCRIPTION
**Pull request by ErwanAliasr1**
Following PR#22463, it appears that some structures & files are missing in dencoder.

This PR did
- inspect all the existing headers to search missing structure
- inspect remaining headers to check which one are providing structures to add in dencoder

This lead to adding 48 types.

### CODE REVIEW

**Comment by liewegas** [comment]
This is the great cleanup.  Just need to make sure make check passes!

File: src/test/encoding/types.h [hunk 1]
```diff
@@ -448,6 +448,11 @@ TYPE(rgw_obj)
 #include "rgw/rgw_log.h"
 TYPE(rgw_log_entry)
 
+#include "rgw/rgw_meta_sync_status.h"
+TYPE(rgw_meta_sync_info)
+TYPE(rgw_meta_sync_marker)
+TYPE(rgw_meta_sync_status)
+
```
Comment (cbodley) [inline src/test/encoding/types.h]: these types are already present just below under `rgw_sync.h`

Comment (ErwanAliasr1) [reply src/test/encoding/types.h]: As those types are declared in rgw_meta_sync_status.h, it seems better to keep them here or remove the one from rgw_sync ?

Comment (cbodley) [reply src/test/encoding/types.h]: agreed!

**Comment by liewegas** [comment]
```
Undefined reference in /home/jenkins-build/build/workspace/ceph-pull-requests/src/test/encoding/ceph_dencoder.cc:135 to `cls_chunk_refcount_read_ret::generate_test_instances(std::__cxx11::list >&)'
```
make sure you 'make ceph-dencoder' and then 'ctest -R readable -V' and 'ctest -R check-generated -V' to verify all is well.

**Comment by ErwanAliasr1** [comment]
@liewegas The latest version passed on my local build, I cross fingers.

**Pull request merged**